### PR TITLE
Fix discard symbol for declaration pattern

### LINKED ISSUE #27086: DeclaredSymbol in IDeclarationPatternOperation shouldn't be null for discard
features/dataflow branch

**Steps to Reproduce**:
Get control flow graph for this code:
```csharp
class C
{
    static void M(object o)
    /*<bind>*/
    {
        if (o is var _)
        {
            return;
        }
    }/*</bind>*/
}
```

**Expected Behavior**:
The declaration pattern should have IDiscardSymbol as the declared symbol.

**Actual Behavior**:
`IDeclarationPatternOperation (**Declared Symbol: null**) `

### PR DESCRIPTION
**Pull request by chborl**
### Customer scenario

Get control flow graph for this code:

```csharp
class C
{
    static void M(object o)
    /*<bind>*/
    {
        if (o is var _)
        {
            return;
        }
    }/*</bind>*/
}
```

### Issue this fixes

#27086

### Workarounds, if any
none

### Risk
low

### Performance impact
low, very small change

### Is this a regression from a previous update?
no

### Test documentation updated?
n/a

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs [hunk 1]
```diff
@@ -1965,6 +1965,11 @@ private IConstantPatternOperation CreateBoundConstantPatternOperation(BoundConst
         private IDeclarationPatternOperation CreateBoundDeclarationPatternOperation(BoundDeclarationPattern boundDeclarationPattern)
         {
             ISymbol variable = boundDeclarationPattern.Variable;
+            if (variable == null && boundDeclarationPattern.VariableAccess.Kind == BoundKind.DiscardExpression)
+            {
+                variable = (IDiscardSymbol)(((BoundDiscardExpression)boundDeclarationPattern.VariableAccess).ExpressionSymbol);
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: Is the cast to `(IDiscardSymbol)` necessary?

Comment (chborl) [reply src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: No, I'll remove it.

**Comment by jinujoseph** [comment]
test windows_release_vs-integration_prtest please

**Comment by jinujoseph** [comment]
Approved to merge for 15.8.Preview5

**Pull request merged**